### PR TITLE
[doc] Separate documentation for tests*.xml files from developer guides

### PR DESCRIPTION
**Pull request by zqzhang**
https://crosswalk-project.org/jira/browse/XWALK-4192

### CODE REVIEW

**Comment by zqzhang** [comment]
@jennycao @Honry @Shao-Feng @haoxli PTAL

File: doc/Web_Test_Suite_Developer_Guide.md [hunk 1]
```diff
@@ -380,135 +380,9 @@ Note:
     </script>
 
 
-## Appendix 1 Tests.full.xml and tests.xml
```
Comment (zqzhang) [inline doc/Web_Test_Suite_Developer_Guide.md]: This `doc/Web_Test_Suite_Developer_Guide.md` documentation looks more like for the WebAPI part only. If no objection, I will rename it to `doc/WebAPI_Test_Suite_Developer_Guide.md` in a separate pull request.

**Comment by zqzhang** [comment]
As sent mail to you all, I will merge this pull request at about 4 pm today if no objection here. Silence will be considered as agree.

**Comment by zqzhang** [comment]
Thanks all for the short offline discussion. We come to conclusion as

1. keep the `Embedding_Test_Suite_Developer_Guide.md`
1. update `Web_Test_Suite_Developer_Guide.md` to be a common test developer guide
1. merge difference in `Cordova_Test_Suite_Developer_Guide.md`, `Web_Runtime_Test_Suite_Developer_Guide.md`, `Web_Test_Suite_SIMD_Developer_Guide.md` into  `Web_Test_Suite_Developer_Guide.md`.

**Comment by jennycao** [comment]
solution B
1. keep developer guide for each components
2. combine the common sections to common mds, such as
# code style
# test suite structure & naming conversion
# tests definitions and organizations

**Comment by jennycao** [comment]
Solution C (from Ling)
1. one over all developer guide to cover common basic guide to develop a simple test. and reference to components developer guide to for specific guides.
2. keep component developer guide to state the specific guides. refer overall developer guide for basic guide.
3. new developer ready steps:
overall developer guide -->component developer guide.

**Comment by zqzhang** [comment]
> overall developer guide -->component developer guide.

Then the `Web_Test_Suite_Developer_Guide.md` serves as the overall developer guide.

**Comment by jennycao** [comment]
merge this PR as we could move the tests.xml samples to this md file instead of long appendix in the developer guide.

**Pull request merged**